server/src/main/java/com/testsigma/service/UploadService.java

```diff
@@ -115,6 +118,32 @@ public void delete(Upload upload) {
     this.uploadRepository.delete(upload);
     publishEvent(upload, EventType.DELETE);
   }
+  public void bulkDelete(Long[] ids, Long workspaceVersionId) throws Exception {
+    Boolean allIdsDeleted = true;
+    TestCaseSpecificationsBuilder builder = new TestCaseSpecificationsBuilder();
+    for (Long id : ids) {
+      List<SearchCriteria> params = new ArrayList<>();
+      List<UploadVersion> uploadVersions = uploadVersionService.findByUploadId(id);
+      //TODO below check in for(uploadVersions)
+      for(UploadVersion uploadVersion: uploadVersions ) {
+        params.add(new SearchCriteria("testData", SearchOperation.EQUALITY, "testsigma-storage:/"+uploadVersion.getPath()));
+        params.add(new SearchCriteria("workspaceVersionId", SearchOperation.EQUALITY, workspaceVersionId));
+        builder.setParams(params);
+        Specification<TestCase> spec = builder.build();
+        Page<TestCase> linkedTestCases = testCaseService.findAll(spec, PageRequest.of(0, 1));
+        if(linkedTestCases.getTotalElements() > 0){
```

Comment:
@Raksha-testsigma Here you are checking whether there are any linked test cases during bulk delete. If there are linked test cases, you are throwing an exception. Don't you need to have the same check in delete method above. The delete method is being called from the controller without any checks

Reply:
@tarun-testsigma Made changes with regard to the exception.